src/containers/ServiceAccountsDropdown/ServiceAccountsDropdown.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { Dropdown, DropdownSkeleton } from 'carbon-components-react';
+
+import {
+  getServiceAccounts,
+  isFetchingServiceAccounts,
+  getSelectedNamespace
+} from '../../reducers';
+import { fetchServiceAccounts } from '../../actions/serviceAccounts';
+
+const itemToElement = ({ text }) => {
```

Comment:
I could have been clearer in my previous comment, when I mentioned reuse I meant that the core of the namespaces dropdown should be refactored to extract the common pieces as a presentational component with the native tooltips for truncated text etc. Then both the namespaces dropdown and this new service accounts dropdown would build on top of that.

Ideally:
- Dropdown (the new presentational component) would
  - provide the loading state using DropdownSkeleton
  - provide these `itemToElement` and `itemToString` implementations by default to the Carbon component
  - automatically transform the `items` prop using `itemStringToObject`

- The `render` function of ServiceAccountsDropdown could look something like this:

```js
  render() {
    return (
      <Dropdown
        {...this.props}
        selectedItem={this.state.selectedItem}
        onChange={this.onChange}
      />
    );
  }
```
where `Dropdown` is the new component I described above

Reply:
Ah, that sounds like the right way to do it 😅 I'll make the change.